sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsIO.java

```diff
@@ -141,76 +171,40 @@ public Read withMaxReadTime(Duration maxReadTime) {
 
       PTransform<PBegin, PCollection<JmsRecord>> transform = unbounded;
 
-      if (maxNumRecords != Long.MAX_VALUE) {
-        transform = unbounded.withMaxNumRecords(maxNumRecords);
-      } else if (maxReadTime != null) {
-        transform = unbounded.withMaxReadTime(maxReadTime);
+      if (getMaxNumRecords() != Long.MAX_VALUE) {
+        transform = unbounded.withMaxNumRecords(getMaxNumRecords());
+      } else if (getMaxReadTime() != null) {
+        transform = unbounded.withMaxReadTime(getMaxReadTime());
       }
 
       return input.getPipeline().apply(transform);
     }
 
     @Override
     public void validate(PBegin input) {
-      checkNotNull(connectionFactory, "ConnectionFactory not specified");
-      checkArgument((queue != null || topic != null), "Either queue or topic not specified");
+      checkNotNull(getConnectionFactory(), "ConnectionFactory not specified");
```

Comment:
Might as well to fix these error messages to be in line with those you have in MqttIO etc?